File: client-data/src/main/java/com/jraska/github/client/users/data/GitHubUserDetailApi.java
package com.jraska.github.client.users.data;

import retrofit2.http.GET;
import retrofit2.http.Path;
import io.reactivex.Single;

import java.util.List;

interface GitHubUserDetailApi {
  @GET("/users/{login}")
  Single<GitHubUserDetail> getUserDetail(@Path("login") String login);

  @GET("/users/{login}/repos?type=all")
  Single<List<GitHubRepo>> getRepos(@Path("login") String login);
}

File: feature-users/src/main/java/com/jraska/github/client/users/model/GitHubUserDetailApi.kt
package com.jraska.github.client.users.model

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

internal interface GitHubUserDetailApi {
  @GET("/users/{login}")
  fun getUserDetail(@Path("login") login: String): Single<GitHubUserDetail>

  @GET("/users/{login}/repos?type=all")
  fun getRepos(@Path("login") login: String): Single<List<GitHubRepo>>
}
